assembl/static2/js/app/pages/surveyAdmin.jsx

```diff
@@ -64,16 +74,38 @@ class SurveyAdmin extends React.Component {
     }
   };
 
+  handleExportLocaleChange = (locale) => {
+    this.setState({ exportLocale: locale });
+  };
+
+  handleTranslationChange = (shouldTranslate) => {
+    this.setState({ translate: shouldTranslate });
+  };
+
   render() {
-    const { section, thematicsHaveChanged } = this.props;
+    const { section, thematicsHaveChanged, debateId, languages } = this.props;
+    const { translate } = this.state;
+    const exportLocale = this.state.exportLocale || (languages && languages[0].locale);
+    const translation = translate && exportLocale ? `?lang=${exportLocale}` : '';
```

Comment:
do you really need translate here ? I would have put it in the local state of `exportSection` component and use it only to know if I have to display the options or not. But it requires to set export locale to `''` when the radio is changed to "don't translate". I guess it's a matter of personal preference here

Reply:
Ok so if I get you right we still want the link to be built in the parent of exportSection but exportSection is in charge of telling if it wants translation and send it up to its parent... The problem in this case is that exportSection will have to do a graphql Query and be connected to the store despite not always needing it.